pkg/frontend/transport/handler.go

```diff
@@ -264,6 +264,7 @@ func (f *Handler) reportSlowQuery(r *http.Request, queryString url.Values, query
 		"host", r.Host,
 		"path", r.URL.Path,
 		"time_taken", queryResponseTime.String(),
+		"header_cache_control", r.Header.Get("Cache-Control"),
```

Comment:
tiny nitpick: it's easier to discover in the code if it's added in the same place other headers are logged. It also won't be duplicated:
https://github.com/grafana/mimir/blob/9ef67470f516f6b2567e5d44e4f21772276a99da/pkg/frontend/transport/handler.go#L422-L429

Reply:
Had to refactor  a little. On the bright side we now have unit tests that check that `-query-frontend.log-query-request-headers` actually works.